synapse/storage/events.py

```diff
@@ -508,36 +502,8 @@ def _persist_events_txn(self, txn, events_and_contexts, backfilled):
         if not events_and_contexts:
             return
 
-        self._store_mult_state_groups_txn(txn, events_and_contexts)
-
-        self._handle_mult_prev_events(
-            txn,
-            events=[event for event, _ in events_and_contexts],
-        )
-
-        for event, _ in events_and_contexts:
-            if event.type == EventTypes.Name:
-                self._store_room_name_txn(txn, event)
-            elif event.type == EventTypes.Topic:
-                self._store_room_topic_txn(txn, event)
-            elif event.type == EventTypes.Message:
-                self._store_room_message_txn(txn, event)
-            elif event.type == EventTypes.Redaction:
-                self._store_redaction(txn, event)
-            elif event.type == EventTypes.RoomHistoryVisibility:
-                self._store_history_visibility_txn(txn, event)
-            elif event.type == EventTypes.GuestAccess:
-                self._store_guest_access_txn(txn, event)
-
-        self._store_room_members_txn(
-            txn,
-            [
-                event
-                for event, _ in events_and_contexts
-                if event.type == EventTypes.Member
-            ],
-            backfilled=backfilled,
-        )
```

Comment:
> Do we have any mechanism to unreject an event yet?
> Do we have any mechanism to asynchronously reject an event that we already accepted?
> It looks like the answer to both question is no in the current code base. The only place we write to the rejections table is in _persist_event, after we've filtered out events that we already have in the db

Therefore I'm sorely tempted to exclude everything except the events and the event_json table until it is demonstrated that it is necessary and safe to do so.